modules/v2/main.tf

```diff
@@ -159,7 +167,17 @@ resource "google_cloud_run_v2_service" "main" {
         }
 
         resources {
-          limits            = containers.value.resources.limits
+          # Setting limits to null when no values are provided prevents a permanent diff
+          # where the provider attempts to remove default values set by the API.
+          limits = length(merge(
+            (try(containers.value.resources.limits.cpu, null) != null ? { cpu = containers.value.resources.limits.cpu } : {}),
+            (try(containers.value.resources.limits.memory, null) != null ? { memory = containers.value.resources.limits.memory } : {}),
```

Comment:
nit: We can calculate this in the locals block so that duplicate calculation is avoided.